### PR TITLE
Safeway

### LINKED ISSUE #27: Safeway
http://local.safeway.com/

### PR DESCRIPTION
**Pull request by whoissteven**
Fixes #27 
Still need to add hours.

### CODE REVIEW

**Comment by whoissteven** [comment]
Hmmm.. So, regex for URL validation isn't correct and still need to fix hours. Will be doing that tomorrow.

File: safeway.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+import scrapy
+import re
+
+from locations.items import GeojsonPointItem
+
+
+class SafewaySpider(scrapy.Spider):
```
Comment (iandees) [inline safeway.py]: Please remove this file.

File: locations/spiders/safeway.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+                path = re.search(locationURL, path)[0].strip('"/>')
+                yield scrapy.Request(
+                    path.strip(),
+                    callback=self.parse_store,
+                )
+
+    def parse_store(self, response):
+
+        properties = {
+            'name': response.xpath('/html/body/main/div[2]/div/section[1]/div[2]/div[1]/h1/span/span[2]/text()').extract(),
```
Comment (iandees) [inline locations/spiders/safeway.py]: These `extract()` calls return arrays. Try replacing them with `extract_first()`.

Comment (whoissteven) [reply locations/spiders/safeway.py]: Deleted accidental upload. Will be pulling changes soon.

File: locations/spiders/safeway.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+                    callback=self.parse_store,
+                )
+
+    def parse_store(self, response):
+
+        properties = {
+            'name': response.xpath('/html/body/main/div[2]/div/section[1]/div[2]/div[1]/h1/span/span[2]/text()').extract_first(),
+            'website': response.request.url,
+            'ref': response.xpath('/html/body/main/div[2]/div/section[1]/div[2]/div[1]/h1/span/span[2]/text()').extract_first(),
+            'opening_hours': " ".join(response.xpath('/html/body/main/div[2]/div/section[1]/div[2]/div[2]/div[1]/div[2]/div[@data-days]').extract_first().split("data-days=\'")[1].split("\'")[0].replace(']','').replace('{','').replace('"day"','').replace('"intervals"','').replace(':','').replace('"',' ').replace(',','').replace('[','').replace('}','').replace('end','close').replace('start','open').split()),
```
Comment (iandees) [inline locations/spiders/safeway.py]: Please correct the opening hours format. See https://github.com/alltheplaces/alltheplaces/pull/254#issuecomment-349794850 for details on the format.

**Pull request merged**